packages/utils/src/cmds/schema/diff/index.ejs

```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE html>
+<meta charset="UTF-8">
+<html>
+    <head>
+        <link rel="stylesheet" href="data:text/css;base64,<%= jsondiffpatchCSS %>">
+
+        <style>
+            th {
+                text-align: left
+            }
+            .left {
+                background-color: #fbb
+            }
+            .right {
+                background-color: #bfb
+            }
+        </style>
+         <%
+            const metaPropsToShow = ['contextPath', 'version', 'revision', 'buildTime']
+            const commitBaseUrl = 'https://github.com/dhis2/dhis2-core/commit/'
```

Comment:
It's from the /info endpoint. I'm unsure how it grabs it, as it's from the backend.

Reply:
@Birkbjo yeah, the problem is that when the image is built outside a git repo (including in our official docker build process) you get this from that endpoint:

```
  "revision": "fatal: Not a git repository (or any of the parent directories): .git",
```

That may be something we want to fix in the backend build process, but for now it's a problem if the backend is built in Docker

Reply:
^ @varl we discussed this an aeon ago (my suggestion I believe was to pass the "build context" into the build rather than presuming anything about the environment in which it's built), might be worth revisiting now that we're adding Docker builds to our Jenkins pipeline?  cc @vilkg

Reply:
I remember. But what we discussed the same thing in a different context: https://github.com/dhis2/dhis2-core/pull/2607/files

The problem back then was when the scripts that bundle the application failed to put out the core's Git revision the build crashed. dhis2/core#2607 solved that by just setting the core revision to "n/a" if it fails.

This is the core pulling in revisions from the build environment: https://github.com/dhis2/dhis2-core/blob/667efb56e42659569e5904d92c492f57e6a4c7ea/dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/system/DefaultSystemService.java#L206

Which is set in the main `pom.xml`: https://github.com/dhis2/dhis2-core/blob/0460082f65826af04d1298d97cc7bca2ef3f47af/dhis-2/pom.xml#L277-L283

That said: would be nice to get this "git agnostic" in both places.

Final note: to get fully git agnostic we would need to change the way the apps are downloaded from `git` to `http`.

Reply:
Yeah it was a slightly different context, you're right.  I agree it would be nice to move to a "stand-alone" build which only relies on the source code in the core repo.

In the meantime, can we make sure the schema diff doesn't break when `revision` is a git error message?

Reply:
This PR solves it so that the build stage 1 can access the Git repo and set the revision properly: https://github.com/dhis2/dhis2-core/pull/3377